sdk/storage/Azure.Storage.DataMovement/src/Shared/DataTransferState.cs

```diff
@@ -103,11 +107,24 @@ public void SetId(string id)
             _id = id;
         }
 
+        /// <summary>
+        /// Gets the status of the transfer
+        /// </summary>
+        /// <returns></returns>
+        public StorageTransferStatus GetTransferStatus()
+        {
+            lock (_statusLock)
+            {
+                return _status;
+            }
+        }
+
         /// <summary>
         /// Sets the completion status
         /// </summary>
         /// <param name="status"></param>
-        public void SetTransferStatus(StorageTransferStatus status)
+        /// <returns>Returns whether or not the status has been changed/set</returns>
+        public bool SetTransferStatus(StorageTransferStatus status)
```

Comment:
Should probably rename `TrySetTransferStatus`

Reply:
done